packages/libraries/client/src/apollo.ts

```diff
@@ -255,25 +255,10 @@ export function hiveApollo(clientOrOptions: HiveClient | HivePluginOptions): Apo
       });
     },
     serverWillStart(ctx) {
-      // `engine` does not exist in v3
-      const isLegacyV0 = 'engine' in ctx;
-
       hive.reportSchema({ schema: ctx.schema });
 
-      if (isLegacyV0) {
-        return {
-          async serverWillStop() {
-            await hive.dispose();
-          },
-        } as any;
-      }
-
       // Works on v3 and v4
-
       return Promise.resolve({
-        async serverWillStop() {
-          await hive.dispose();
-        },
```

Comment:
Removing this seems to be a breaking change?

Reply:
It is a breaking change

Reply:
we shouldn't remove it...

Reply:
Is it actually a breaking change ? Since the dispose is now automatically done on process exit ?

Reply:
Could be. It's `serverWillClose` of Apollo Server that controls when the server is about to be closed. It may have extra steps before shutting down the server, like waiting inflight requests to be completed.
That's why I prefer to keep Apollo plugin as close to Apollo as possible in this case, but have autoDispose for Yoga and the rest.

Reply:
Ok, let me rollback to the privous implementation then